autoPyTorch/datasets/base_dataset.py

```diff
@@ -231,3 +244,9 @@ def replace_data(self, X_train: BASE_DATASET_INPUT, X_test: Optional[BASE_DATASE
         if X_test is not None and self.test_tensors is not None:
```

Comment:
I think this if statement should only be `if X_test is not None`, that is the behavior I would expect no?

Reply:
Hey, actually this X_test is coming from the fit dictionary and the test_tensors are coming from the base dataset. Basically, the X_test would be the preprocessed data(if it's small enough) coming from the early preprocessing node. So this replace data function would put the preprocessed X_train and X_test from the fit dictionary into the data loader else it would use from the backend.

Reply:
i see, then this is resolved